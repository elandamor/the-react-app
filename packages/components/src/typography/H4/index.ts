import {
  shouldForwardProp,
  space,
  styled,
  StyledSystemProps,
  typography,
} from '@paisidevs/tra-theme';

/**
 * @render react
 * @name H4 component
 * @description Heading level 4.
 * @example
 *  <H4>Heading</H4>
 */

const H4 = styled('h4', { shouldForwardProp })<StyledSystemProps>`
  ${space};
  ${typography};
`;

H4.defaultProps = {
  color: 'text.default',
  fontSize: [5, 6],
};

export default H4;
