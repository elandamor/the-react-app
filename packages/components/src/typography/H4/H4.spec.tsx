import { render } from '@testing-library/react';
import React from 'react';
import H4 from './index';

describe('H4', () => {
  it('renders without crashing', () => {
    const { container } = render(<H4>Heading</H4>);
    expect(container.firstChild?.nodeName.toLowerCase()).toBe('h4');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correct text', () => {
    const { getByText } = render(<H4>Heading</H4>);
    expect(getByText('Heading')).toBeTruthy();
  });
});
