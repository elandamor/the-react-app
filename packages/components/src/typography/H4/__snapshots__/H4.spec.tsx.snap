// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H4 renders without crashing 1`] = `
.emotion-0 {
  font-size: 32px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 48px;
  }
}

<h4
  class="emotion-0"
>
  Heading
</h4>
`;
